trace/kafka/component.go

```diff
@@ -0,0 +1,161 @@
+package kafka
+
+import (
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/Shopify/sarama"
+	"github.com/beatlabs/patron/encoding"
+	"github.com/beatlabs/patron/encoding/json"
+	patronErrors "github.com/beatlabs/patron/errors"
+	"github.com/beatlabs/patron/log"
+	"github.com/opentracing/opentracing-go"
+)
+
+// RequiredAcks is used in Produce Requests to tell the broker how many replica acknowledgements
+// it must see before responding.
+type RequiredAcks int16
+
+const (
+	// NoResponse doesn't send any response, the TCP ACK is all you get.
+	NoResponse RequiredAcks = 0
+	// WaitForLocal waits for only the local commit to succeed before responding.
+	WaitForLocal RequiredAcks = 1
+	// WaitForAll waits for all in-sync replicas to commit before responding.
+	WaitForAll RequiredAcks = -1
+)
+
+const fieldSetMsg = "Setting property '%v' for '%v'"
+
+// Builder gathers all required and optional properties, in order
+// to construct a Kafka AsyncProducer.
+type Builder struct {
```

Comment:
1. We should be more specific `AsyncBuilder`. 
2. A component in Patron is something else, so we should rename the file to kafka.go.

Reply:
Should this code be part of the already existing `kafka.go` and `kafka_test.go` which handle the actual kafka logic, or something like `builder.go` and `builder_test.go`?

Reply:
builder.go then. :+1:

Reply:
Done, thanks. I rebased with the `master` branch and applied the other suggestions as well